tests/Aspire.Hosting.Tests/Publishing/PublishingActivityProgressReporterTests.cs

```diff
@@ -514,6 +514,135 @@ public async Task CompleteInteractionAsync_ProcessesUserResponsesCorrectly()
         Assert.Equal("user-response", promptResult.Data?.Value);
     }
 
+    [Fact]
+    public async Task CalculateAggregatedState_WithNoTasks_ReturnsCompleted()
+    {
+        // Arrange
+        var reporter = new PublishingActivityProgressReporter(_interactionService);
+        var step = await reporter.CreateStepAsync("Test Step", CancellationToken.None);
+
+        // Act
+        var aggregatedState = step.CalculateAggregatedState();
+
+        // Assert
+        Assert.Equal(CompletionState.Completed, aggregatedState);
```

Comment:
In the moment from creating the step to creating the first task wouldn't this make the aggregated state complete, even though its just getting started?

Reply:
We only call CalculateAggregatedState() in DisposeAsync